lib/event_sourcery/event_processing/esp_runner.rb

```diff
@@ -22,7 +22,7 @@ def start!
         with_logging do
           start_processes
           listen_for_shutdown_signals
-          wait_till_shutdown_requested
+          wait_till_shutdown_requested_or_process_terminates
```

Comment:
Instead of only waiting for shutdown signals, also monitor the child processes. If they fail it's time to quit.